### PR TITLE
Functional Wallet container

### LINKED ISSUE #2891: Decrediton making calls after shutdown requested
We need to clean requests that happens sporadically before closing decrediton.

### PR DESCRIPTION
**Pull request by amassarwi**
This diff fixes #2891, as we had intervals to fetch politeia's token inventory every minute & wallet GetPeerInfo call every ten seconds but we didn't cleanup the intervals on container's unmount.

This commit includes a bit more than that, as I took the opportunity to refactor the `Wallet` container to functional approach.

Changes:
 - Functional `containers/Wallet.jsx`.
 - Get rid of `WalletContainer` connector & introduce `useWallet` hook.
 - Partially get rid of `Theming` connector and replace it with `useTheming` hook.
 - Use `useEffect` to set the intervals & to clean them up on unmount.

---

Closes #2891
Part of #2438

### CODE REVIEW

**Review by thi4go — APPROVED** [review]
tACK and looking good to me mate, nice work towards #2438

**Comment by amassarwi** [comment]
@vctt94 rebased mate

**Review by vctt94 — APPROVED** [review]
Thanks. LGTM!

@bgptr can you see if the issue you've seen is fixed by this PR?

File: app/containers/Wallet.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const piInvInterval = setInterval(() => {
+      if (politeiaEnabled) {
+        compareInventory();
+      }
+    }, 60000);
+
+    // Get peer info every 10 seconds, so we can no if there are no available
+    // peers.
+    const peerInfoInterval = setInterval(() => {
+      if (politeiaEnabled) {
```
Comment (bgptr) [inline app/containers/Wallet.jsx:55]: Is `politeiaEnabled` have to be `true` here for getting peer info?

**Comment by bgptr** [comment]
@amassarwi @vctt94 Looking great. I just left an inline question.

File: app/containers/Wallet.jsx (hunk 1 above)
Comment (vctt94) [reply app/containers/Wallet.jsx:55]: No it shouldn't good catch!

Comment (vctt94) [reply app/containers/Wallet.jsx:55]: @amassarwi

Comment (amassarwi) [reply app/containers/Wallet.jsx:55]: that's was the original condition, will get rid of it

**Pull request merged**